### PR TITLE
Added auto commit issue changes and oracle data types

### PR DESCRIPTION
**Pull request by dnwick**

### CODE REVIEW

File: components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.rdbms/src/main/java/org/wso2/carbon/event/output/adapter/rdbms/RDBMSEventAdapter.java [hunk 1]
```diff
@@ -324,6 +324,7 @@ public void executeDbActions(Object message)
                     stmt = con.prepareStatement(executionInfo.getPreparedUpdateStatement());
                     populateStatement(map, stmt, executionInfo.getUpdateQueryColumnOrder());
                     int updatedRows = stmt.executeUpdate();
+                    con.setAutoCommit(false);
```
Comment (lasanthafdo) [inline components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.rdbms/src/main/java/org/wso2/carbon/event/output/adapter/rdbms/RDBMSEventAdapter.java]: Shouldn't con.setAutoCommit(false) be called before we prepare the statement and execute it. If we set autoCommit to false just before committing, at the time of doing stmt.executeUpdate(), autoCommit is still set to true. Wouldn't that mean that at the time of executing the statement, there is still a possibility of autoCommit being triggered?

File: components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.rdbms/src/main/java/org/wso2/carbon/event/output/adapter/rdbms/RDBMSEventAdapter.java [hunk 2]
```diff
@@ -339,6 +340,7 @@ public void executeDbActions(Object message)
                     stmt = con.prepareStatement(executionInfo.getPreparedInsertStatement());
                     populateStatement(map, stmt, executionInfo.getInsertQueryColumnOrder());
                     stmt.executeUpdate();
+                    con.setAutoCommit(false);
```
Comment (lasanthafdo) [inline components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.rdbms/src/main/java/org/wso2/carbon/event/output/adapter/rdbms/RDBMSEventAdapter.java]: Same as above, shouldn't autoCommit false happen earlier?

File: components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.rdbms/src/main/java/org/wso2/carbon/event/output/adapter/rdbms/RDBMSEventAdapter.java [hunk 3]
```diff
@@ -424,6 +426,7 @@ public void createTableIfNotExist(String tableName)
             try {
                 if (!tableExists) {
                     stmt.executeUpdate(executionInfo.getPreparedCreateTableStatement());
+                    con.setAutoCommit(false);
```
Comment (lasanthafdo) [inline components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.rdbms/src/main/java/org/wso2/carbon/event/output/adapter/rdbms/RDBMSEventAdapter.java]: Same as above, shouldn't autoCommit false happen earlier?

**Comment by mohanvive** [comment]
Yes, That is posible. Asked Damith to change these.. Damith please check the siddhi code on how it handled..

**Comment by dnwick** [comment]
Ya it should be like that. I missed it. Will make the change.

On Thu, Apr 9, 2015 at 12:49 PM, Mohanadarshan Vivekanandalingam <
notifications@github.com> wrote:

> Yes, That is posible. Asked Damith to change these.. Damith please check
> the siddhi code on how it handled..
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/wso2/carbon-event-processing/pull/96#issuecomment-91133994>
> .
>



-- 
Software Engineer
WSO2 Inc.; http://wso2.com
<http://www.google.com/url?q=http%3A%2F%2Fwso2.com&sa=D&sntz=1&usg=AFQjCNEZvyc0uMD1HhBaEGCBxs6e9fBObg>
lean.enterprise.middleware

mobile: *+94728671315*

**Pull request merged**